src/Mod/Path/PathScripts/PathOp.py

```diff
@@ -579,3 +726,129 @@ def addBase(self, obj, base, sub):
+            do.Shape = objShape
+            do.purgeTouched()
+        if self.showDebugObjects:
+            self.tmpDebugGrp.addObject(do)
+
+    # Template methods to be overwritten in child operation classes as needed
+    def __getstate__(self):
+        '''__getstat__(self) ... called when receiver is saved.
+        Can safely be overwritten by subclasses.'''
+        return None
+
+    def __setstate__(self, state):
+        '''__getstat__(self) ... called when receiver is restored.
+        Can safely be overwritten by subclasses.'''
+        return None
+
+    def initOperation(self, obj):
+        '''initOperation(obj) ... implement to create additional properties.
+        Should be overwritten by subclasses.'''
+        pass  # pylint: disable=unnecessary-pass
+
+    def _initOpAttributes(self, obj):
+        '''_initOpAttributes(obj) ... Set class attributes that must be set
+        at new operation instantiation and also upon document restoration.
+        Should be overwritten by sub-class.'''
+        pass
+
+    def opFeatures(self, obj):
+        '''opFeatures(obj) ... returns the OR'ed list of features used and supported by the operation.
+        The default implementation returns "FeatureTool | FeatureDepths | FeatureHeights | FeatureStartPoint"
+        Should be overwritten by subclasses.'''
+        # pylint: disable=unused-argument
+        return FeatureTool | FeatureDepths | FeatureHeights | FeatureStartPoint | FeatureBaseGeometry | FeatureFinishDepth | FeatureCoolant
+
+    def opPropertyDefinitions(self):
+        '''opPropertyDefinitions() ... Return operation-specific list
+        of property defininitions stored as tuples in the format of:
+        (prototpe, name, group name, tooltip).
+        Should be overwritten by sub-class.'''
+        return []
+
+    def opPropertyEnumerations(self):
+        '''opPropertyEnumerations() ... returns a dictionary of property
+        enumeration lists as {'property_name': [option1, option2]}.
+        This method should be overwritten by the operation sub-class.'''
+        # Enumeration lists for App::PropertyEnumeration properties
+        return {}
+
+    def opPropertyDefaults(self, obj, job):
+        '''opPropertyDefaults(obj, job) ...
+        This method should be overwritten by the operation sub-class.
+        Default property values should be assigned to class dictionary
+        attribute as self.propDefaults['property_name'] = value'''
+        return {}
+
+    def opSetEditorModes(self, obj):
+        '''opSetEditorModes(obj) ...
+        Process operation-specific changes to properties visibilities.
+        This method may be overwritten by subclasses.'''
+        pass
+
+    def opOnDocumentRestored(self, obj):
+        '''opOnDocumentRestored(obj) ... implement if an op needs special handling like migrating the data model.
+        Should be overwritten by subclasses.'''
+        pass  # pylint: disable=unnecessary-pass
```

Comment:
FYI: Originally I tried to stick to a certain vocabulary:
_"can safely be overwritten"_  .... a subclass is free to overwrite if there is a desire
_"should be overwritten"_  ... a subclass is expected to overwrite this function, the return value or its side effects are required for proper functionality

Doesn't look like that intent communicated the intention though ;)